connection.go

```diff
@@ -76,6 +76,8 @@ type connection struct {
 	// The peer has everything. This can occur due to a special message, when
 	// we may not even know the number of pieces in the torrent yet.
 	peerHasAll bool
+	// number pieces completed
+	peerCompleted int
```

Comment:
I think this value can be inferred from the peerHas Bitmap or peerHasAll. It should be a method.